### PR TITLE
test: Updates for Dual Region and testing

### LINKED ISSUE #1304: Add Support for Dual Region Buckets
**FR:** Support dual region buckets.

The method for creating a bucket should accept a string with 2 regions concatenated with a '+' (e.g. `US-EAST1+US-WEST1`).

**System tests:**
- Verify a dual region bucket with specified regions can be created
- Verify a dual region bucket's `location` metadata is the specified dual region provided
- Verify a dual region bucket's `locationType` metadata is `dual-region`

[Internal] For details and coordination: [go/gcs-dpe-dual-region-milestones](http://go/gcs-dpe-dual-region-milestones)

### PR DESCRIPTION
**Pull request by sydney-munro**
Fixes #1304  ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java [hunk 1]
```diff
@@ -1731,7 +1731,7 @@ public Long getMetageneration() {
 
   /**
    * Returns the bucket's location. Data for blobs in the bucket resides in physical storage within
-   * this region.
+   * this region or regions.
    *
    * @see <a href="https://cloud.google.com/storage/docs/bucket-locations">Bucket Locations</a>
```
Comment (frankyn) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java:1736]: Does dual-region have a link that we can direct folks to our is it just the bucket-locations doc? 

cc: @danielbankhead

Comment (danielbankhead) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java:1736]: That will be here: https://cloud.google.com/storage/docs/locations

(the updates are currently in staging, pending for release)

Comment (sydney-munro) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java:1736]: it appears the link `https://cloud.google.com/storage/docs/bucket-locations` redirects to `https://cloud.google.com/storage/docs/locations` I can update this link.

**Comment by danielbankhead** [comment]
LGTM, may want another Java expert to review. Then, we'll merge once https://github.com/googleapis/python-storage/pull/748 is unblocked

**Pull request merged**